### PR TITLE
Postgres 16 support updates

### PR DESCRIPTION
**Pull request by danieltprice**
Release note:
https://neon-next-git-dprice-postgres-16-support-neondatabase.vercel.app/docs/release-notes

Some additional updates to:
- Migration docs
- neondb database (call it a "ready-to-use" database instead of "default database") . There's nothing special about other than it allows you to make an initial connection.
- Update to the Postgres extensions table

### CODE REVIEW

**Review by duskpoet — APPROVED** [review]

File: content/docs/extensions/pg-extensions.md [hunk 1]
```diff
@@ -5,84 +5,85 @@ redirectFrom:
...
+| [pg_embedding](https://github.com/neondatabase/pg_embedding)             | 0.3.50           | 0.3.5  | 0.3.5            | Install with `CREATE EXTENSION embedding;`                                                                         |
+| [pg_graphql](https://github.com/supabase/pg_graphql)               | 1.1.0                 | 1.1.0                  |  1.1.0                  |                                                                                                                   |
+| [pg_hashids](https://github.com/iCyberon/pg_hashids)                | 1.2.1                | 1.2.1                  |  1.2.1                  |                                                                                                                  |
+| [pg_hint_plan](https://github.com/ossc-db/pg_hint_plan)             | 1.5                  | 1.5                  |  1.5                  |                                                                                                                   |
```
Comment (bayandin) [inline content/docs/extensions/pg-extensions.md:42]: We have different versions of `pg_hint_plan` for each Postgres version:
- 1.4.1 for Postgres 14
- 1.5.0 for Postgres 15
- 1.6.0 for Postgres 16

https://github.com/neondatabase/neon/blob/dce91b33a4ce24b1526ef1c39a95761cb0d7da2b/Dockerfile.compute-node#L421-L437

File: content/docs/extensions/pg-extensions.md [hunk 2]
```diff
@@ -5,84 +5,85 @@ redirectFrom:
...
-| unit                     | [7.7](https://github.com/df7cb/postgresql-unit)                 | [7.7](https://github.com/df7cb/postgresql-unit)                  |                                                                                                                    |
-| uuid-ossp                | [1.1](https://www.postgresql.org/docs/14/uuid-ossp.html)        | [1.1](https://www.postgresql.org/docs/15/uuid-ossp.html)         | Double-quote the extension name when installing: `CREATE EXTENSION "uuid-ossp"`                                    |
-| xml2                     | [1.1](https://www.postgresql.org/docs/current/xml2.html)        | [1.1](https://www.postgresql.org/docs/current/xml2.html)         |                                                                                                                    |
+| Extension                | Version (Postgres 14)                               | Version (Postgres 15)  | Version (Postgres 16)                              |Notes                                                                                                                |
```
Comment (bayandin) [inline content/docs/extensions/pg-extensions.md:12]: Some of the extensions don't support Postgres 16 yet:
- `pg_jsonschema`
- `pg_graphql`
- `pg_tiktoken`
- `pgx_ulid`

**Review by bayandin — CHANGES_REQUESTED** [review]

File: content/docs/extensions/pg-extensions.md (hunk 2 above)
Comment (bayandin) [reply content/docs/extensions/pg-extensions.md:12]: Also `timescaledb` doesn't support Postgres 16 yet

Comment (bayandin) [reply content/docs/extensions/pg-extensions.md:12]: And we don't install `pg_embedding` to Postgres 16

File: content/docs/extensions/pg-extensions.md (hunk 1 above)
Comment (danieltprice) [reply content/docs/extensions/pg-extensions.md:42]: Thank you. Updated.

File: content/docs/extensions/pg-extensions.md (hunk 2 above)
Comment (danieltprice) [reply content/docs/extensions/pg-extensions.md:12]: Thank you. Updated.

File: content/release-notes/2023-09-15-storage-and-compute.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+![Postgres 16 Create project](/docs/relnotes/postgres_16.png)
+
+To migrate data from an existing Neon project to one created with Postgres 16, refer to the dump and restore procedure described in [Import data from another Neon project](/docs/import/import-from-neon).
+
+As with any database migration, always test thoroughly before migrating production systems or applications. Also, we recommend familiarizing yourself with the changes in Postgres 16, especially those affecting compatibility. For information about those changes, please refer to the official [Postgres Release 16 documentation](https://www.postgresql.org/docs/16/release-16.html).
+
+### plv8 extension upgrade
+
+The `plv8` extension was updated from 3.1.5 to 3.1.8 in Postgres 16. For more information about Postgres extensions supported by Neon and how to install them, see [Postgres extensions](/docs/extensions/pg-extensions).
```
Comment (bayandin) [inline content/release-notes/2023-09-15-storage-and-compute.md:13]: Should we drop this section?
We didn't have Postgres 16 before, so we never had plv8 3.1.5 for Postgres 16 at all.

**Review by bayandin — DISMISSED** [review]

**Review by bayandin — APPROVED** [review]

**Pull request merged**